xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,140 @@
+            content1 = file1.read()
+            content2 = file2.read()
+
+        return content1 == content2
+
+    except (FileNotFoundError, PermissionError):
+        print(f"Error opening file {file_path1} or {file_path2}")
+        return False
+
+
+def are_arrays_equal(array1, array2) -> bool:
+    if len(array1) != len(array2):
+        return False
+
+    for i in range(len(array1)):
+        if array1[i] != array2[i]:
+            return False
+
+    return True
+
+
+arg_input_xml: str = "--input-taco-path"
+arg_output_xml: str = "--output-taco-path"
+
+chrono_patterns = [
+    "^The taco parse function took [0-9]+ milliseconds to run$",
+    "^The xml write function took [0-9]+ milliseconds to run$",
+    "^The protobuf write function took [0-9]+ milliseconds to run$"
+]
+
+
+if __name__ == "__main__":
+    # TODO: This will need to be extracted as more attributes and tests are added
+    attribute: Attribute_Tests = Attribute_Tests(
+        attribute_name="can_fade",
+        attribute_path="../test_cases/can_fade_tests",
+        tests=["_is_correct", "_is_different_file", "_is_wrong_output"],
+        output_result={
+            "_is_correct": ['Loading taco pack ../test_cases/can_fade_tests/can_fade_is_correct.xml'],
+            "_is_different_file": ['Loading taco pack ../test_cases/can_fade_tests/can_fade_is_different_file.xml'],
+            "_is_wrong_output": ['WRONG']
+        },
+        expect_output_to_be_equal={
+            "_is_correct": True,
+            "_is_different_file": True,
+            "_is_wrong_output": False,
+        },
+        expect_files_to_be_equal={
+            "_is_correct": True,
+            "_is_different_file": False,
+            "_is_wrong_output": True,
+        }
+    )
+
+    # Ensure that the test output directory is created
+    if not os.path.exists(attribute.attribute_path + "-output/"):
+        try:
+            os.makedirs(attribute.attribute_path + "-output/")
+        except OSError as e:
+            print(f"Error: {e}")
+
+    for test in attribute.tests:
+        arg1 = attribute.attribute_path + "/" + attribute.attribute_name + test + ".xml"
+        arg2 = attribute.attribute_path + "-output/" + attribute.attribute_name + test + ".xml"
+
+        result = run_xml_converter(arg_input_xml, arg1, arg_output_xml, arg2)
+        if result is not None:
+            # Remove lines about how long it took to run
+            split_result = result.split("\n")
+            filtered_array = []
+            for line in split_result:
+                match_found = False
+                for pattern in chrono_patterns:
+                    if re.search(pattern, line) or line == "":
+                        match_found = True
+                        break
+                if not match_found:
+                    filtered_array.append(line)
+
+            output_test: bool = True
+            file_test: bool = True
```

Comment:
This can just be one variable of `all tests passed` or something like that. And if any test fails then it can set it to false.

Reply:
I like that plan. If any test is false it already prints an error so we don't need to know which test failed later.